helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java

```diff
@@ -53,6 +54,9 @@
   public static Set<String> UNHEALTHY_STATES =
       ImmutableSet.of(HelixDefinedState.DROPPED.name(), HelixDefinedState.ERROR.name());
 
+  static final String UNHEALTHY_PARTITION = "UNHEALTHY_PARTITION";
+  static final String PARTITION_INITIAL_STATE_FAIL = "PARTITION_INITIAL_STATE_FAIL";
```

Comment:
NIT: could we have comments here explaining the definitions and contexts. For example: why is it a "fail" for partitions in initial states?

Reply:
Done.